### DISCUSSION TITLE
Allow Shuffle for all Music tracks

### DISCUSSION THREAD

**Discussion by vk2him** [music-assistant/support#405] [Feature requests and ideas]
The ability to play/add to queue **all** music tracks doesn't seem to be an option. I'd like to select all my music tracks and shuffle play them, similar to how it works on the iPhone's Music App.

**Comment by marcelveldt** [comment]
You mean the entire tracks collection ?
This feature already exists for collections like albums, artisttracks and playlists but not for an entire toplevel.

**Comment by vk2him** [comment]
Yes, entire tracks - thanks

**Comment by neurove** [comment]
I would like to push this feature request up.

Requesting the following usability improvements:
- "Play" or "Add all tracks to queue" button at toplevel.
- "Select all/none" button when selecting multiple tracks/artists/etc.
- "Shuffle" and a "Repeat" toggle button on the play area  (currently it takes too many clicks to find shuffle/repeat in the settings).

Thanks again for such a great software :)

**Reply by OzGav** [reply]
You can already select all although there is an open issue to be resolved https://github.com/music-assistant/hass-music-assistant/issues/821

The next version due in December has a new front end which has the shuffle and repeat buttons on the main screen.

**Reply by neurove** [reply]
Thanks. looking forward to the december release
